test/compilable/test19038.d

```diff
@@ -0,0 +1,11 @@
+module test19038;
+
+void test()
+{
+    if ([[cast(const) "x"]] == [["x"]])
+    {
+    }
+    if ([[cast(const) 5]] == [[5]])
+    {
+    }
```

Comment:
It would be better to type out variables instead of casting literals.
```D
const a = "a";
```
Will do it

Reply:
fixed